.github/workflows/test.yml

```diff
@@ -0,0 +1,75 @@
+name: test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  msrv-check:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu-latest, macos-latest]
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+
+      - name: Get MSRV
+        run: |
+          RUST_MSRV="$(cat near-abi-client/Cargo.toml | sed -n 's/rust-version *= *"\(.*\)"/\1/p')"
+          echo "RUST_MSRV=$RUST_MSRV" >> $GITHUB_ENV
+
+      - name: "Install ${{ env.RUST_MSRV }} toolchain (MSRV)"
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ env.RUST_MSRV }}
+
+      - name: Cargo check
+        run: cargo check -p near-abi-client
+
+  tests:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu-latest, macos-latest]
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+
+      - name: Install `wasm32-unknown-unknown`
+        run: rustup target add wasm32-unknown-unknown
+
+      - name: Run tests
+        run: cargo test --workspace --verbose
+
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+
+      - name: Check Formatting
+        run: cargo fmt -p near-abi-client -p near-abi-client-macros -p near-abi-client-impl -- --check
+
+      - name: Check Clippy
+        run: cargo clippy -p near-abi-client -p near-abi-client-macros -p near-abi-client-impl --tests -- -Dclippy::all
```

Comment:
Are the explicit package definitions useful here?

Reply:
Just saw https://github.com/near/near-sdk-abi/pull/5#discussion_r1060295789, got it.